### PR TITLE
Feature 289 layout view clickable in preview

### LINKED ISSUE #289: layout view clickable in preview

### PR DESCRIPTION
**Pull request by lauriLukkarinen**
Resolves #289

### CODE REVIEW

File: src/components/preview/components/page-preview-button.tsx [hunk 1]
```diff
@@ -40,19 +44,32 @@ class PagePreviewButton extends React.Component<Props, State> {
...
   }
 
   /**
    * Render
    */
   public render() {
+    const { view, selectedView, onResize } = this.props;
+    const { mouseOver } = this.state;
+    const selected = selectedView?.id === view.id;
+    const selectedHighlight = mouseOver || selected ?
```
Comment (anttileppa) [inline src/components/preview/components/page-preview-button.tsx:58]: Could this be moved to styles? Or at lest generalized in one place, it's currently repeated in all of the components

File: src/components/preview/components/page-preview-frame-layout.tsx [hunk 2]
```diff
@@ -202,6 +228,33 @@ class PagePreviewFrameLayout extends React.Component<Props, State> {
...
+   * Event handler for mouse out
+   * 
+   * @param event react mouse event
+   */
+  private onMouseOut = (event: React.MouseEvent) => {
+    event.stopPropagation();
+  }
+
+  /**
+   * Event handler for mouse over
```
Comment (anttileppa) [inline src/components/preview/components/page-preview-frame-layout.tsx:251]:
```suggestion
   * Event handler for mouse click
```

File: src/components/preview/components/page-preview-linear-layout.tsx [hunk 3]
```diff
@@ -196,6 +224,33 @@ class PagePreviewLinearLayout extends React.Component<Props, State> {
...
+   * Event handler for mouse out
+   * 
+   * @param event react mouse event
+   */
+  private onMouseOut = (event: React.MouseEvent) => {
+    event.stopPropagation();
+  }
+
+  /**
+   * Event handler for mouse over
```
Comment (anttileppa) [inline src/components/preview/components/page-preview-linear-layout.tsx:247]:
```suggestion
   * Event handler for mouse click
```

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/components/preview/components/page-preview-component.tsx [hunk 4]
```diff
@@ -71,85 +74,135 @@ class PagePreviewComponent extends React.Component<Props, State> {
...
+      handleLayoutProperties,
+      onViewClick
+    } = this.props;
+
+    if (!view) {
       return null;
     }
 
-    switch (this.props.view.widget) {
+    const zIndex = layer ? layer + 1 : 1;
```
[the remark this answers is not in the document]
Comment (lauriLukkarinen) [reply src/components/preview/components/page-preview-component.tsx]: Layer is now resolved for components via getLayer method, that returns current layer

File: src/components/preview/components/page-preview-button.tsx (hunk 1 above)
Comment (lauriLukkarinen) [reply src/components/preview/components/page-preview-button.tsx]: boxShadow definition moved to styles

**Pull request merged**